pkg/console/operator/status.go

```diff
@@ -0,0 +1,295 @@
+//   group: config.openshift.io
+//   kind:  ClusterOperator
+//   name:  console
+//
+// Status Condition Types
+// Status conditions should not be set to a "default" state when the operator starts up.
+// Instead, set a status explicitly only when the state is known.  This is because
+// the various status conditions may have been set on previous runs of the sync loop, or
+// possibly even by a previous operator container.
+//
+// Available = the operand (not the operator) is available
+//    example: the console resources are stabilized.
+//    example: the console is not yet in a functional state, one or
+//      more resources may be missing.
+// Progressing = the operator is trying to transition operand state
+//    example: during the initial deployment, the operator needs to create a number
+//      of resources.  the console (operand) is likely in flux.
+//    example: the assigned route .spec.host changes for the console.  the oauthclient
+//      must be updated, as must the configmap, etc.  numerous resources are in flux,
+//      the operator reports progressing until the resources stabilize
+// Failing = the operator (not the operand) is failing
+//    example: The console operator is unable to update the console config with
+//      a new logoutRedirect URL. The operator is failing to do its job, however the
+//      console (operand) may or may not be functional (see Available above).
+//
+//
+// Status Condition Reason & Message
+// Reason:  OperatorSyncLoopError
+// Message: "The operator sync loop was not completed successfully"
+//
+//
+
+// Lets transition to using this, and get the repetition out of all of the above.
+func (c *consoleOperator) SyncStatus(operatorConfig *operatorsv1.Console) (*operatorsv1.Console, error) {
+	logConditions(operatorConfig.Status.Conditions)
+	updatedConfig, err := c.operatorConfigClient.UpdateStatus(operatorConfig)
+	if err != nil {
+		return nil, fmt.Errorf("status update error: %v \n", err)
+	}
+	return updatedConfig, nil
+}
+
+// Outputs the condition as a log message based on the detail of the condition in the form of:
+//   Status.Condition.<Condition>: <Bool>
+//   Status.Condition.<Condition>: <Bool> (<Reason>)
+//   Status.Condition.<Condition>: <Bool> (<Reason>) <Message>
+//   Status.Condition.<Condition>: <Bool> <Message>
+func logConditions(conditions []operatorsv1.OperatorCondition) {
+	logrus.Println("Operator.Status.Conditions")
+	for _, condition := range conditions {
+		buf := bytes.Buffer{}
+		buf.WriteString(fmt.Sprintf("Status.Condition.%s: %s", condition.Type, condition.Status))
+		hasMessage := condition.Message != ""
+		hasReason := condition.Reason != ""
+		if hasMessage && hasReason {
+			buf.WriteString(" |")
+			if hasReason {
+				buf.WriteString(fmt.Sprintf(" (%s)", condition.Reason))
+			}
+			if hasMessage {
+				buf.WriteString(fmt.Sprintf(" %s", condition.Message))
+			}
+		}
+		logrus.Println(buf.String())
+	}
+}
+
+// setStatusCondition
```

Comment:
This is slightly less verbose than doing it all inline. I've used funcs below to try to abstract away the noise even more.

Reply:
I feel like all the helpers might actually be adding code without really making things simpler, but I won't ask you to change it :)

Reply:
Prefer going back to inline?  See PR #140. We could make the manifest updates, etc there